Project organization

sofiaburkow opened the pull request:
Major changes:
- Organized the project using the Cookiecutter template for data science
- Used Poetry to handle dependencies and virtual environment.
- Organized the original "training.py" into several scripts.

Notes:
- Some scripts may be redundant (like evaluate.py and predict.py). We can look into deleting unnecessary files later.
- I have included some empty folders and files, like "tests". If they will not be used by others later we can delete them then.
- Did not manage to fix all bugs in time. I got to the step of running the features.py file ("python model_training/features.py") where I encountered some errors. 

How to test:
- Follow the instructions in the README.

annaneliaa commented:
Looks good! Maybe we can add a bit more instructions on running model training in the readme. It's not super clear what the correct order of executing the scripts is etc. for outsiders

sofiaburkow commented:
Have updated README with instructions on how to train the model (the full pipeline).  Also fixed the aforementioned bug and removed some unnecessary files. 

The Makefile is still there, however, I am unsure whether we need it. Also, I have not tested that automatic versioning still works.
